### ISSUE TITLE
Error in installer.sh after apps detection

### ISSUE THREAD

**Issue by c2b2pss** [winapps-org/winapps#71]
Everything runs fine except that at the end of the installer process, it detects  and configures apps and then I get 3 lines of errors:

```

Configuring Access... Finished.
  Configuring Excel... Finished.
  Configuring OneNote... Finished.
  Configuring Outlook... Finished.
  Configuring PowerPoint... Finished.
  Configuring Publisher... Finished.
  Configuring Word... Finished.
/home/user/.local/share/winapps/detected: line 4: You: command not found
/home/user/.local/share/winapps/detected: line 5: At: command not found
/home/user/.local/share/winapps/detected: line 6: syntax error near unexpected token `('
/home/user/.local/share/winapps/detected: line 6: `+         $Name = (Get-Item $Exe).VersionInfo.FileDescription.Trim() -r ...'
```

Here is copy of the file "detected" in /home/user/.local/share/winapps/detected
LINES RIGHT AT BEGINNING:

```
NAMES=()
ICONS=()
EXES=()
You cannot call a method on a null-valued expression.
At \\tsclient\home\.local\share\winapps\ExtractPrograms.ps1:215 char:9
+         $Name = (Get-Item $Exe).VersionInfo.FileDescription.Trim() -r ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull

NAMES+=("")
EXES+=("C:\Program Files\cassowary\cassowary.exe")
ICONS+=("AAABAAEAICAQ1QAAAADoAgAAFgAAACgAAAAgAAAAQAAAAAEABAAAAAAAgAIAAAAAAAAAAA>
NAMES+=("Epson Scan 2 Launcher")
EXES+=("C:\Program Files (x86)\epson\Epson Scan 2\Core\es2launcher.exe")
ICONS+=("AAABAAEAICAQ1QAAAADoAgAAFgAAACgAAAAgAAAAQAAAAAEABAAAAAAAgAIAAAAAAAAAAA>
NAMES+=("Epson Scan 2 Utility")
EXES+=("C:\Program Files (x86)\epson\Epson Scan 2\Core\es2utility.exe")
ICONS+=("AAABAAEAICAQZQAAAADoAgAAFgAAACgAAAAgAAAAQAAAAAEABAAAAAAAgAIAAAAAAAAAAA>
NAMES+=("Microsoft Excel")
```

**Comment by LDprg** [comment]
@c2b2pss ist this fixed with #139? Just clone the repo and try again.

**Comment by c2b2pss** [comment]
Hi,

Thanks for attending to this.

Please give me couple of days to try this out 👍

**Comment by KernelGhost** [comment]
The `detected` file at `~/.local/share/winapps/detected` is created by running `ExtractPrograms.ps1` within Windows. Unfortunately, I did not make any changes to the PowerShell script in #139.

The PowerShell script works by enumerating entries within `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\`, where `(Default)` entries contain the path to the corresponding `.exe` files. There are some entries, such as `table30.exe` and `setup.exe` for which the `(Default)` value is `(value not set)`, and these are skipped by the script (i.e. `Where-Object {$_."(default)" -ne $null}`).

Reading the error log provided, it appears `cassowary.exe` is the culprit, given that the command `$Name = (Get-Item $Exe).VersionInfo.FileDescription.Trim() -replace "  "," "` appears to fail. Whilst `(Get-Item "C:\Program Files\7-Zip\7zFM.exe").VersionInfo.FileDescription.Trim() -replace "  "," "` produces the output `7-Zip File Manager`, for example, `(Get-Item "C:\Program Files\cassowary\cassowary.exe").VersionInfo.FileDescription` fails to produce any output.

This is a bug in the PowerShell script, as the script fails to check whether the `FileDescription` property is set. This bug can be corrected by defaulting to using the executable file name (i.e. "cassowary.exe") in cases where `FileDescription` is not set.

**Comment by KernelGhost** [comment]
I have tested and confirmed the fix in #151.

```
NAMES+=("cassowary")
EXES+=("C:\Program Files\cassowary\cassowary.exe")
ICONS+=("AAABAAEAICAQAAAAAADoAgAAFgAAACgAAAAgAAAAQAAAAAEABAAAAAAAgAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAACAAAAAgIAAgAAAAIAAgACAgAAAgICAAMDAwAAAAP8AAP8AAAD//wD/AAAA/wD/AP//AAD///8AAAAAAAAAAAAAAAAAAAAAAAAAZmZniId3dwAAAAAAAAAARmZmaIiIiIiIiIh2ZERAAEZmZmiIiIiIiIiId3d3ZABGZmZoiIiIiIiIiHd3d3YARmZmeIiIiIiIiIh3d3d0AGZmZniIiIiIiIiId3d3dABmZmZ4iIiIiIiIiHd3d3QAZmZmeIiIiIiIiIh3d3dkAGZmZniIiIiIiIiId3d3YABmZmZ3d3d4iIiIiHd3d2AAZmZmd3d3d3d3d3d3d3dgBGYBMzMwRmZnd3d3d3d3YARnO7u7swAAAAAARmZnd2AEe7u7u7uwAAAAB3AAAGdgBHu7u7u7swAAAAD3AABnYAY7u7u7u7MAAAAAf3AAZkAGO7u7u7uzAAAAAAfwAGZABju7u7u7swAAAAAIgABmQAY7u7u7u7MAAAAH9wAAZgAGO7u7u7u7u7sAB3AAAGYABju7u7u7uzO7AAAAAABmAAY7u7u7u7sAOwAAAAAAZgBGO7u7u7u7M7sAAAAABGYARzu7u7u7u7uzAAAAAARmAEc7u7u7u7uzMAAAAAAGZgBnO7u7u7EAAAAAAAAABmQAZzu7u7uwAAAAAAAAAAZkAEY7u7u7sAAAAAAAAAAGZAAAO7u7u7AAAAAAAAAABmAAADu7u7szAEREAAAAAAZgAAADM7szMAAAAAAAAEREQADgAAAH4AAAB8AAAAHAAAAAwAAAAMAAAADAAAAAwAAAAMAAAADAAAABwAAAAcAAAAGAAAABgAAAAYAAAAGAAAABgAAAAYAAAAGAAAABgAAAA4AAAAOAAAADgAAAAwAAAAMAAAADAAAAAwAAAAMAAAADAAAAA8AAAAfADAAH4B//Bw==")
```
